mirrord/layer/src/tcp_steal.rs

```diff
@@ -230,8 +228,20 @@ impl TcpHandler for TcpStealHandler {
             return Ok(());
         }
 
-        let steal_type = if self.http_ports.contains(&original_port) && let Some(filter_str) = self.http_filter.take() {
-            FilteredHttp(request_port, Filter::new(filter_str)?)
+        let steal_type = if self.http_filter_settings.ports.contains(&original_port) {
+            match self.http_filter_settings.filter {
+                LayerHttpFilter::None => All(request_port),
+                LayerHttpFilter::HeaderDeprecated(ref header) => {
+                    FilteredHttp(request_port, Filter::new(header.clone())?)
```

Comment:
We could also use the new type if the user specified the config in the old way, no?

Reply:
We could, but I want to let the users have the ability to configure if old protocol or new essentially.